components/src/core/Drawer/DrawerNavItem.tsx

```diff
@@ -403,7 +405,7 @@ const DrawerNavItemRender: React.ForwardRefRenderFunction<HTMLElement, DrawerNav
 
     // Combine the classes to pass down the the InfoListItem
     const infoListItemClasses = {
-        root: clsx(defaultClasses.infoListItemRoot, classes.infoListItemRoot),
+        root: ripple && hasAction ? undefined : clsx(defaultClasses.infoListItemRoot, classes.infoListItemRoot),
```

Comment:
Can you clarify why you are only applying these styles when there is no ripple and action?

Reply:
When an InfoListItem has a click event associated with it, we have to render a ListItemButton inside it...this button needs to have the padding rather than the root element or the ripple will be inside of the padding.

Reply:
However, I think we should still apply the user supplied classes here (classes.infoListItemRoot). Unless we go to having a single shared class as suggested above.